### ISSUE TITLE
Maintenance – Community

### ISSUE THREAD

**Issue by signalwerk** [namespace-ee/react-calendar-timeline#84]
Hey @mariusandra 
I'm not sure who's having push permission to this repo, but I see that there are a couple pull requests hanging and there is a community effort to bring the component forward. It's hard to keep the community together if everyone is working in its own repo but don't push upstream again because people see no activity in the repo anymore. 

Therefore my request; I completely know it's a hassle to keep maintaining a project if day-jobs and stuff are eating your time. But could you please give someone else (or a few) push access so we could also keep the momentum without bothering you? It's always good if the "original" is the pot to bring all the commits back together. I know technically every fork could also do that but people usually find only the "original". What do you think? 

Personally I use the component in a payed job and I work every few days on that project. I would be willing to invest a bit work in helping (with others) to maintain this repo during 2017. @meengit works with me and we worked on the multiple selection together (https://github.com/namespace-ee/react-calendar-timeline/pull/71) we also have 1-2 other changes we could commit later that year. We are building a copy past function in our software to duplicate multiple items.

**Comment by schwermer** [comment]
@signalwerk  -  you're absolutely right, our company (and I'm sure others) have ended up forking the project due to lack of activity.  Several critical issues (eg #76, #64) have fixes in PRs but they have not been merged, not to mention all the great features that other have added.  

I would also be available to help maintain this project, it is a critical piece of our application and we want to see it continue to improve!

**Comment by madisvain** [comment]
Hello! Sorry about this delay. Yes this is a critical part of our application also. We will be reviewing the pull requests and answering questions latest:  1st of July. Hopefully even next week.

And of course we can discuss somebody coming on board also to have push/commit permissions to this library so we can build it faster. I will discuss this today and get back to you beginning of next week.

**Comment by madisvain** [comment]
@schwermer @signalwerk I have talked with Marius and have decided that we will add Stefan as a contributor to this project so he will also have the push and commit access to maintain and release new versions. As he took initiative in this and I think would be a step in the right direction and building a community.

Ourselves we also use this component as a very important part of our projects. We will be putting in new effort towards react-calendar-timeline in the upcoming weeks but I understand that this is too long of a delay.

So Stefan let me know if you have any questions. Otherwise great to have somebody on board with us and maybe we can discuss future goals of react-calender-timeline together at some point and set a roadmap where we would like it to go in the future.

**Comment by signalwerk** [comment]
What a terrible timing of me... First I raise my voice and then I'm a bit off the grid... Sorry guys. I'm next week back in the office and plan time to review a couple PRs. I think for the first release after some month I would plan a new branch for version 0.12.0 before that goes to master 

@schwermer I would love to see you in the loop and help to bring this repo back to a state where you could also take the original fork again. 

we didn't have an update during the last 5 months so give me/us a bit time for the work. thx

**Comment by nicocrm** [comment]
Hi guys, joining the discussion as we have picked the timeline in our app and would like to contribute some improvements - this is more of a heads up at this point because they are not quite ready for a pull request but I was glad to see some activity on the project!  Thank you!

**Comment by bkniffler** [comment]
How is the state? There is still some critical PRs open and I'd love to see them merged :)

**Comment by signalwerk** [comment]
@bkniffler I'm right now a bit in a hurry with other stuff but I'm work on it. I took one of the worst bugs first (#76 fixed by @schwermer) and made now my first new release. Version 0.11.2 lives right now in https://github.com/signalwerk/react-calendar-timeline/tree/hotfix/group-ordering 

since it's my first release I would like to get first an OK of @madisvain or @schwermer before I merge it to master. I try to do it right rather than to do it quick. I also started to introduce a test-framework (jest & enzyme) to have the bug covered by a test. I'll not write for each PR a test but if we start to try to write tests when we write stuff we will probably end in a better code quality.

**Comment by signalwerk** [comment]
@madisvain to do my first release I also need permission for npm (https://docs.npmjs.com/cli/owner) could you please add me? signalwerk is also my name on npm

**Comment by lnagel** [comment]
@signalwerk Added!

**Comment by madisvain** [comment]
@signalwerk congrats for the first release! I'll close this issue for now. We'll also be helping you out more regarding this and getting the development on track. Please don't hesitate to ask for our help in some of these issues and we'll be putting effort into this library also in the upcoming months to bring it further.

**Comment by signalwerk** [comment]
@madisvain thanks a lot!
maybe you noticed that I made a little mistake. version 0.11.2 was never published on npm so it's 0.11.3 now. too avoid that in the future and to do the release task automatically I would like to setup a travis-ci deploy task. so my final request regarding user rights:

could you please add a deploy key for me to the repo? with a deploy key travis-ci is able to push the tags, npm and other stuff... it's not my personal ssh-key it's a fresh key and is never used before. the idea is that only travis-ci will use it and it then only has access to that repo. you can find the instruction how to add it to the repo her: https://developer.github.com/guides/managing-deploy-keys/#deploy-keys 
should take you 2 min to do it.

here is the public key for travis:

```
ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAACAQC84CcWY7qDC74g/cZDuo513axKlFX0IPchVjcXTA9vlZpGLF9/cseTQxtBFqeMH8Axxr2pec/upegylRQ/ZzQvL1wBmpe3Z0ez7A38OKUXouN87BCRy7COp6WxDEoocQpQbcWC+/VGE2ci4gEttaqSkR3m63fcrwBtPIKv0bxc3MaqZ5jZdOhoEwUMWzZ8pdVmlEy7bZu7+l48yOhNHrKkedp1wTZA0dhuoATacaC0ctWOCM5eZ6e5/Ty0NRBgb4Xg+HhX81EjXRY4S901o5XgUXhR7m2nXSCXY4k3a/w6fI/ckxG9IjNwwwEDLmkZIcAnqTHJ6zqt+UhOnuTAP0cFtNwJaH+0L0GYEGSRNwitMrdqYZA+FAKn6zdvxMo3vcNHc9NQ06/XF6cNITDj/vzSNYxtNYTiS5yhef8Po3vI3KyidEc8mBYOn3dYQYC8qm6obkM3VpXNgAFcebyBUuUQLyw0zfqZu3PlvTyviobGRssJpMyvQ4+cotD3E1KaADeI5vd9i8Ks4x6LzxlbtfkyTO0KHN3w7G6nZTRE4OgY9CjQQLDi3nje1GzzqA1lS65aRhM89FMwSjQmoaTKAoiKl/HjcfW7vRVoSDnhM8SjOjxLMepizSeHPDNR32xgsQ8LGd7vPlWUjv6u/L2E6Eae2VF3WixER7+GGWNVxPfHWw== sh@signalwerk.ch
```

With that I could also automate the gh-page so that the demo page of the component could be hosted on github and you don't need to host it on your personal server anymore.

**Comment by lnagel** [comment]
@signalwerk Public repos should not need a deploy key in Travis, there is no such section even in the public travis settings area.

I just enabled this repo on travis, check it out here: https://travis-ci.org/namespace-ee/react-calendar-timeline

**Comment by nicocrm** [comment]
Do you guys need help in setting some of this up?  Would love to see more updates on this repo, we are running a custom version of the timeline in our app - it has proven super useful and I'd love to give something back :)

This is something we did with custom renderers:

![image](https://cloud.githubusercontent.com/assets/64349/26602863/d9ac40ea-4552-11e7-9864-d5fcb49c6883.png)

![image](https://cloud.githubusercontent.com/assets/64349/26602919/08c80346-4553-11e7-8d66-a08c54a604a8.png)

**Comment by madisvain** [comment]
Hei guys! So we have still not been able to get started from our side. I think we need to expand the community a bit for this library to move forward.

@signalwerk what are your plans currently and also @schwermer @nicocrm what do you propose how shall we continue? We are open to expanding the community but I think if its going to be more people we should somehow start discussing the features that make it into the project and also the main goals of this library.

**Comment by nicocrm** [comment]
Thanks Madis, I picked the library in part because it is relatively lightweight compared to some of the options out there and I think it would be great to keep it that way, but add some extension points so that users can further configure it to fit their needs (and also of course, include the PR that bring bug fixes and improve the documentation).  Thank you for your work on this!

**Comment by mariusandra** [comment]
Hey everyone!

I discussed with @madisvain and we agreed to work on this project again for the foreseeable future, contributing a few hours each week.

I started today with some PRs and cleaning up the package structure, package.json, build scripts, etc. 

Some of the bigger changes so far:
- Merged two simple PRs (#96 and #101)
- Fixed travis - now tests work
- Removed the UMD builds (/build/dist folder). Please let me know if anyone was still using them. I don't think they should be in git anyway.
- Removed the "modules" folder from git (now just "src" is here) and added a "prepublish" script that automatically builds the library before pushing. Now the publishing process is just "npm version minor/patch; npm publish" (and git push)
- Replaced the old demo with a new webkit-based one (/demo/demo.js). I still need @madisvain to change the github page from the gh-pages branch to the /docs folder, into which the demo is now build
- Released v0.12.0 with the two merged PRs and the new project structure to make sure everything works as intended. It seems to have worked.

I will be around in the next days responding to issues, cleaning up PRs and maintaining as well as I can, at least a few hours each week. So expect more updates to come! That's it for today.

**Comment by signalwerk** [comment]
@madisvain @mariusandra Thank you guys for your constant efforts to bring the project forward. I couldn't spend too much time the last couple weeks here but I try to get back to it. 

@mariusandra your suggestions sounds really good! The Travis CI integration I suggested a couple days ago was intended to publish automatically to the 'gh-pages' branch from within Travis-CI. But for that Travis needs access. I did it with an encrypted deploy key. Here you see how I set it up for an other project of mine: https://github.com/signalwerk/colorlab/blob/master/deploy_doc.sh

**Comment by mariusandra** [comment]
@signalwerk That's cool! I'll try to get that working at some point. It's indeed better to do this than to pollute the repository with big .js bundles. Thanks! :)

**Comment by signalwerk** [comment]
@mariusandra it's pretty simple if you have a key set in github. I can do it (then I need the public ssh in the github) or if you like to do it yourself just hit me if you need help.

**Comment by mariusandra** [comment]
I released 0.13.0 with most of the PRs. I also added the deploy script, so as soon as @madisvain approves the ssh key, it'll be set to auto deploy.

Thank you everyone for your contributions! @goooseman @nicocrm @meikoudras @meengit @jmerriweather @signalwerk @bkniffler

*(I really hope I didn't miss anyone... :))*

The updated demo is here: http://namespace.ee/react-calendar-timeline/

In the next days I'll investigate the remaining PRs and issues... and then spend some effort on making the timeline faster/smoother as well.
